pkg/client/restclient/config.go

```diff
@@ -278,16 +299,29 @@ func InClusterConfig() (*Config, error) {
 // IsConfigTransportTLS returns true if and only if the provided
 // config will result in a protected connection to the server when it
 // is passed to restclient.RESTClientFor().  Use to determine when to
-// send credentials over the wire.
+// send credentials over the wire. Any non-https host will return false.
+// All Hosts in Config must be either TLS enabled or not.
 //
 // Note: the Insecure flag is ignored when testing for this value, so MITM attacks are
 // still possible.
-func IsConfigTransportTLS(config Config) bool {
-	baseURL, _, err := defaultServerUrlFor(&config)
+func IsConfigTransportTLS(config Config) (bool, error) {
+	hosts, _, err := defaultServerUrlFor(&config)
+	allHttps := 0
+	// TODO Propagate errors from defaultServerUrlFor
```

Comment:
What is the problem in those tests?  Maybe add a comment to the TODO explaining the problem if that is the case.